srcpkgs/memtest86+/patches/06.memtest86+-5.01-port-io.patch

```diff
@@ -0,0 +1,41 @@
+--- test.c.orig
++++ test.c
+@@ -14,7 +14,6 @@
+ #include "stdint.h"
+ #include "cpuid.h"
+ #include "smp.h"
+-#include <sys/io.h>
+
+ extern struct cpu_ident cpu_id;
+ extern volatile int    mstr_cpu;
+@@ -29,6 +29,30 @@
+ ulong rand(int me);
+ void poll_errors();
+ 
++static inline unsigned char inb(unsigned short port)
++{
++	unsigned char val;
++	asm volatile ("inb %1,%0" : "=a" (val) : "dN" (port));
++	return val;
++}
++
++static inline unsigned char inb_p(unsigned short int port)
++{
++	unsigned char val;
++	asm volatile ("inb %w1,%0\noutb %%al,$0x80":"=a" (val):"Nd" (port));
++	return val;
++}
++
++static inline void outb(unsigned char val, unsigned short port)
++{
++	asm volatile ("outb %0,%1" : : "a" (val), "dN" (port));
++}
++
++static inline void outb_p(unsigned char val, unsigned short int port)
++{
++	asm volatile ("outb %b0,%w1\noutb %%al,$0x80": :"a" (val),"Nd" (port));
++}
```

Comment:
I'd rather you copy the whole thing from `glibc`, instead of using partly glibc's style and partly musl's.

Please add a comment at the top of the patch explaining that musl doesn't have the `_p` functions.

Reply:
Please check if everything is correct now.